### DISCUSSION TITLE
How to generate the correct tags layout?

### DISCUSSION THREAD

**Discussion by Novecento99** [gijzelaerr/python-snap7#528] [Q&A] [answered]
Hi,
I need to access a PLC db in a convenient way. 
Since my db is composed of multiple structs made of different data types, I really need to use the automatic function that parse the db layout.

Well, I've tried different things from the TIA portal but it seems I'm really unable to extract the correct view...

this is taken from this repo example:

> rc_if_db_1_layout_bu = """
> 
> 4   RC_IF_ID    INT
> 6   RC_IF_NAME  STRING[16]
> 
> 24.0    LockAct         BOOL    # interlocked or not
> 24.1    GrpErr          BOOL    # indicate error
> 24.2    RuyToStart      BOOL
> """

when I try to export it, the most similar situation I can retrieve is this:

> 	Static											
> 												
> 	PieroAppoggiato	Bool	0.0	false	TRUE	False	True	True	True	False		
> 	lole	Byte	1.0	16#0	---	False	True	True	True	False	

how to extract the correct layout format ?

thankyou

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. The DB layout format expects: `offset name type`. Create this manually from your TIA Portal export.
